src/enemy.rs

```diff
@@ -0,0 +1,279 @@
+                    if Some(true) == spawner.is_elite && current_wave.need_wave_spawn == false {
+                        continue
+                    }
+                    if let Ok(p_t) = player.get_single_mut() {
+                        let is_left = thread_rng().gen_range(0, 2);
+                        let is_up = thread_rng().gen_range(0, 2);
+                        let mut enemy_batch: Vec<(EnemyBundle, SpriteBundle, SteeringBundle, Name)> = Vec::new();
+
+                        for i in 1..thread_rng().gen_range(ENEMY_BATCH_SIZE.min, ENEMY_BATCH_SIZE.max) {
+                            let (mut m_x, mut m_y, mut m_z): (f32, f32, f32);
+                            if is_left == 1 {
+                                m_x = thread_rng().gen_range(p_t.translation.x - SPAWN_DISTANCE.max, p_t.translation.x - SPAWN_DISTANCE.min);
+                            } else {
+                                m_x = thread_rng().gen_range(p_t.translation.x + SPAWN_DISTANCE.min, p_t.translation.x + SPAWN_DISTANCE.max);
+                            }
+
+                            if is_up == 1 {
+                                m_y = thread_rng().gen_range(p_t.translation.y - SPAWN_DISTANCE.max, p_t.translation.y - SPAWN_DISTANCE.min);
+                            } else {
+                                m_y = thread_rng().gen_range(p_t.translation.y + SPAWN_DISTANCE.min, p_t.translation.y + SPAWN_DISTANCE.max);
+                            }
+
+                            m_z = p_t.translation.z;
+                            enemy_batch.push(
+                                (EnemyBundle::new(spawner.health.clone(), spawner.damage.clone(), spawner.rewards.clone()),
+                                 SpriteBundle {
+                                     transform: Transform::from_translation(Vec3::new(m_x, m_y, m_z)),
+                                     texture: spawner.texture.clone(),
+                                     ..default()
+                                 },
+                                 SteeringBundle {
+                                     host: SteeringHost {
+                                         position: Vec2::new(m_x, m_y),
+                                         max_velocity: spawner.max_velocity,
+                                         max_force: spawner.max_force,
+                                         mass: spawner.mass,
+                                         ..default()
+                                     },
+                                 },
+                                 Name::new(spawner.name.clone() + &i.to_string())));
+                            if let Some(true) = spawner.is_elite {
+                               break;
+                            }
+                        }
+                        commands.spawn_batch(enemy_batch)
+                    }
+                }
+            }
+            if current_wave.need_wave_spawn == true {
```

Comment:
:thinking:

Reply:
i can remove the `== true` if you are about it